core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -235,11 +235,11 @@ object SparkSubmit extends CommandLineUtils with Logging {
   private[deploy] def prepareSubmitEnvironment(
       args: SparkSubmitArguments,
       conf: Option[HadoopConfiguration] = None)
-      : (Seq[String], Seq[String], Map[String, String], String) = {
+      : (Seq[String], Seq[String], SparkConf, String) = {
     // Return values
     val childArgs = new ArrayBuffer[String]()
     val childClasspath = new ArrayBuffer[String]()
-    val sysProps = new HashMap[String, String]()
+    val sparkConf = new SparkConf()
```

Comment:
Hi, @vanzin .
Is it intentionally loading default config? Previously, it wasn't in [line 340](https://github.com/apache/spark/pull/19519/files#diff-4d2ab44195558d5a9d5f15b8803ef39dL340).

Reply:
Yes. Becase this conf will now be exposed to apps (once I change code to extend `SparkApplication`), the conf needs to respect system properties.

In fact the previous version should probably have done that too from the get go.

Reply:
Thanks!